cmd/private-org-peribolos-sync/main.go

```diff
@@ -0,0 +1,233 @@
+	if err != nil {
+		logger.WithError(err).Fatal("couldn't get the list of org/repos that promote official images")
+	}
+
+	peribolosRepos := generateRepositories(gc, orgRepos, logger)
+	peribolosConfigByOrg := peribolosConfig.Orgs[o.destOrg]
+	peribolosConfigByOrg.Repos = peribolosRepos
+	peribolosConfig.Orgs[o.destOrg] = peribolosConfigByOrg
+
+	out, err := yaml.Marshal(peribolosConfig)
+	if err != nil {
+		logrus.WithError(err).Fatalf("%s failed to marshal output.", o.peribolosConfig)
+	}
+
+	if err := ioutil.WriteFile(o.peribolosConfig, out, 0666); err != nil {
+		logrus.WithError(err).Fatal("Failed to write output.")
+	}
+}
+
+func generateRepositories(gc gitHubClient, orgRepos map[string]sets.String, logger *logrus.Entry) map[string]org.Repo {
+	peribolosRepos := make(map[string]org.Repo)
+
+	for orgName, repos := range orgRepos {
+		for repo := range repos {
+			logger.WithFields(logrus.Fields{"org": orgName, "repo": repo}).Info("Processing repository details...")
+
+			fullRepo, err := gc.GetRepo(orgName, repo)
+			if err != nil {
+				logger.WithError(err).Fatal("couldn't get repo details")
+			}
+
+			peribolosRepos[fullRepo.Name] = org.PruneRepoDefaults(org.Repo{
+				Description:      &fullRepo.Description,
+				HomePage:         &fullRepo.Homepage,
+				Private:          &fullRepo.Private,
+				HasIssues:        &fullRepo.HasIssues,
+				HasProjects:      &fullRepo.HasProjects,
+				HasWiki:          &fullRepo.HasWiki,
+				AllowMergeCommit: &fullRepo.AllowMergeCommit,
+				AllowSquashMerge: &fullRepo.AllowSquashMerge,
+				AllowRebaseMerge: &fullRepo.AllowRebaseMerge,
+				Archived:         &fullRepo.Archived,
+				DefaultBranch:    &fullRepo.DefaultBranch,
+			})
+		}
+	}
+
+	return peribolosRepos
+}
+
+func getOrgReposWithOfficialImages(releaseRepoPath string, includeRepos map[string]sets.String) (map[string]sets.String, error) {
+	ret := make(map[string]sets.String)
+
+	callback := makeCallback(includeRepos, ret)
+	if err := config.OperateOnCIOperatorConfigDir(filepath.Join(releaseRepoPath, config.CiopConfigInRepoPath), callback); err != nil {
+		return ret, fmt.Errorf("error while operating in ci-operator configuration files: %v", err)
+	}
+
+	return ret, nil
+}
+
+// makeCallback returns a function usable for OperateOnCIOperatorConfigDir that picks the orgs/repos
+// that promotes official images and the one that are in whitelist.
+func makeCallback(includeRepos, orgReposPicked map[string]sets.String) func(*api.ReleaseBuildConfiguration, *config.Info) error {
+	return func(c *api.ReleaseBuildConfiguration, i *config.Info) error {
+
+		// skip this repo unless it's in the whitelist
+		if !promotion.BuildsOfficialImages(c) {
+			if repos, _ := includeRepos[i.Org]; !repos.Has(i.Repo) {
+				return nil
+			}
+		}
+
+		if _, ok := orgReposPicked[i.Org]; !ok {
+			orgReposPicked[i.Org] = sets.NewString(i.Repo)
+		} else {
+			orgReposPicked[i.Org].Insert(i.Repo)
+		}
```

Comment:
Could this be a simple `Union` call? `orgReposPicked[i.Org]` should return a zero value of a sets.String when `i.Org` is not in the map, and the zero value should be fine to use in a `Union`...